### PR TITLE
Profiles swift compile times

### PR DESCRIPTION
**Pull request by ashfurrow**
Following advice from here: https://medium.com/@johnsundell/improving-swift-compile-times-ee1d52fb9bd#.lhtrt8wc0

Not sure about the pod changes, will look into them tomorrow.

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -372,7 +372,7 @@ SPEC CHECKSUMS:
   Extraction: 612cf0866f74d4c0dd616677ff24146efa200958
   FBSDKCoreKit: d2aaed5e9ab7d8d6301c533376a1fbff1cf3deb5
   FBSDKLoginKit: 699ff169080e3072de4b9b0faca90bf23dc36deb
-  FBSnapshotTestCase: 094f9f314decbabe373b87cc339bea235a63e07a
+  FBSnapshotTestCase: '094f9f314decbabe373b87cc339bea235a63e07a'
```
Comment (orta) [inline Podfile.lock]: this is just because people have different ruby versions. I wonder if CP should consider amending the YML to make these consistent

**Comment by alloy** [comment]
You’ll get no objections from me 👍 But maybe wait to merge till 3.2.0 is released?

**Comment by ashfurrow** [comment]
That sounds prudent @alloy. I've got #2197 also waiting for 3.2.0. Should I create a post-3.2.0 branch to merge both of these into for now?

**Comment by alloy** [comment]
Nah, 3.2.0 is probably going to be submitted today.

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**